core/common/src/main/java/alluxio/util/executor/ControllableQueue.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util.executor;
+
+import java.util.PriorityQueue;
+
+/**
+ * A controllable queue is a priority queue with {@link DelayNode}s.
+ * It supports tick the time forward and pop nodes whose delay is smaller than the past time.
+ *
+ * @param <T> the type of the value of {@link DelayNode}
+ */
+public class ControllableQueue<T> {
+  private long mPastTime;
+  private PriorityQueue<DelayNode<T>> mQueue;
+
+  /**
+   * Constructs a new {@link ControllableQueue}.
+   */
+  public ControllableQueue() {
+    mPastTime = 0;
+    mQueue = new PriorityQueue<>((n1, n2) -> {
+      long diff = n1.getDelay() - n2.getDelay();
+      return diff > 0 ? 1 : diff < 0 ? -1 : 0;
+    });
+  }
+
+  /**
+   * @return whether this queue is empty
+   */
+  public boolean isEmpty() {
+    return mQueue.isEmpty();
+  }
+
+  /**
+   * @return the value of head element
+   */
+  public T getPeakValue() {
+    return mQueue.peek().getValue();
+  }
+
+  /**
+   * @return the delay of head element
+   */
+  public long getPeakDelay() {
```

Comment:
`PeakDelay` sounds like it should be the highest delay, but it's actually returning the lowest

Reply:
Changed to `getHeadDelay` to avoid the miss-understanding. Thanks